src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ShareACaseUserListPreparer.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.*;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.ProfessionalUsersRetriever;
+
+@Slf4j
+@Service
+public class ShareACaseUserListPreparer implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private ProfessionalUsersRetriever professionalUsersRetriever;
+
+    public ShareACaseUserListPreparer(ProfessionalUsersRetriever professionalUsersRetriever) {
+        this.professionalUsersRetriever = professionalUsersRetriever;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_START
+               && callback.getEvent() == Event.SHARE_A_CASE;
+
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callbackStage,
+                                                        Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        return mapToAsylumCase(callback, professionalUsersRetriever.retrieve());
+
+    }
+
+    private PreSubmitCallbackResponse<AsylumCase> mapToAsylumCase(Callback<AsylumCase> callback,
+                                                                  ProfessionalUsersResponse usersResponse) {
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        DynamicList dynamicList;
+
+        final List<Value> values =
+            usersResponse.getProfessionalUsers()
+                .stream()
+                .filter(professionalUser ->
+                    professionalUser.getIdamStatus().equalsIgnoreCase("ACTIVE"))
```

Comment:
In future we can use the parameter in PRD API that allows a filtered response.  This is good for the core version

Reply:
:+1: